### PR TITLE
Fancy prompt

### PR DESCRIPTION
**Pull request by TibbTobb**

### CODE REVIEW

File: fancy_prompt/fancy_prompt.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+Enhanced prompt support for UndoDB.
+Contributors: Mark Williamson, Toby Lloyd Davies
+Copyright (C) 2019 Undo Ltd
+'''
+
+from __future__ import absolute_import, division
+import re
+import datetime
+from undodb.debugger_extensions import udb
+import gdb
```
Comment (barisione) [inline fancy_prompt/fancy_prompt.py]: This needs rearranging.

File: fancy_prompt/fancy_prompt.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+    def __init__(self):
+        super(PromptColour, self).__init__('prompt-color', gdb.COMMAND_SUPPORT)
+
+    @staticmethod
+    def invoke(argument, from_tty):
+        global colour_enabled
+        if argument == 'on':
+            colour_enabled = True
+        elif argument == 'off':
+            colour_enabled = False
```
Comment (barisione) [inline fancy_prompt/fancy_prompt.py]: Maybe raise an exception if you get something else.

File: fancy_prompt/fancy_prompt.py [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+def undodb_get_extent():
+    '''
+    Get the event log extent of the current debuggee.
+
+    Returns: (bb_start, bb_end) tuple of start and end bbcounts.
+    '''
+    undodb_state = undodb_get_state()
+    if undodb_state == STATE_DEFER or undodb_state == STATE_NONE:
+        return None
+    return udb.get_event_log_extent()
```
Comment (barisione) [inline fancy_prompt/fancy_prompt.py]: What happens if you don't do the check first?

Comment (TibbTobb) [reply fancy_prompt/fancy_prompt.py]: state_none: udbpy.comms.NotRunningError
state_defer: udbpy.comms.NotConnectedError

File: fancy_prompt/fancy_prompt.py (hunk 2 above)
Comment (TibbTobb) [reply fancy_prompt/fancy_prompt.py]: Done

File: fancy_prompt/fancy_prompt.py (hunk 1 above)
Comment (TibbTobb) [reply fancy_prompt/fancy_prompt.py]: Done

**Pull request merged**